src/hooks/useAuth.tsx

```
@@ -77,16 +83,26 @@ export function AuthProvider({ children }: { children: ReactNode }) {
     setUser(null);
   }, []);
 
   useEffect(() => {
-    /* eslint-disable react-hooks/set-state-in-effect -- Restore browser-only auth state after hydration so the server and first client render match. */
-    setUser(getUserFromStorage());
-    setLoading(false);
-    /* eslint-enable react-hooks/set-state-in-effect */
+    const restoreUser = async () => {
+      /* eslint-disable react-hooks/set-state-in-effect -- Restore browser-only auth state after hydration so the server and first client render match. */
+      // setUser(getUserFromStorage());
+      setUser(await getUserData());
```

ESLint

[error] Error: Cannot access variable before it is declared

`getUserData` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/src/hooks/useAuth.tsx:100:21
   98 |       /* eslint-disable react-hooks/set-state-in-effect -- Restore browser-only auth state after hydration so the server and first client render match. */
   99 |       // setUser(getUserFromStorage());
> 100 |       setUser(await getUserData());
      |                     ^^^^^^^^^^^ `getUserData` accessed before it is declared
  101 |       setLoading(false);
  102 |       /* eslint-enable react-hooks/set-state-in-effect */
  103 |     };

/home/jailuser/git/src/hooks/useAuth.tsx:137:3
  135 |
  136 |   
> 137 |   const getUserData = useCallback(async () => {
      |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 138 |     setLoading(true);
      | ^^^^^^^^^^^^^^^^^^^^^
> 139 |     setError(null);
      …
      | ^^^^^^^^^^^^^^^^^^^^^
> 150 |
      | ^^^^^^^^^^^^^^^^^^^^^
> 151 |   }, []);
      | ^^^^^^^^^^ `getUserData` is declared here
  152 |
  153 |   const loginWithCode = useCallback(async (code: string) => {
  154 |     setLoading(true);

(react-hooks/immutability)